rust/kernel/types.rs

```diff
@@ -238,6 +238,14 @@ impl<T> Opaque<T> {
     pub fn get(&self) -> *mut T {
         UnsafeCell::raw_get(self.0.as_ptr())
     }
+
+    /// Gets the value behind `this`.
+    ///
+    /// This function is useful to get access to the value without creating intermediate
+    /// references.
+    pub const fn raw_get(this: *const Self) -> *mut T {
```

Comment:
Would it better be a `*mut Self -> *mut T`?

Reply:
I dont know. I copied what [`UnsafeCell::raw_get`](https://doc.rust-lang.org/std/cell/struct.UnsafeCell.html#method.raw_get) does.

Reply:
maybe @nbdd0121 @bjorn3 know more?

Reply:
This type should be defined as `UnsafeCell<MaybeUninit<T>>`.

Reply:
Then we should change that, but I do not want to put it into this patch series.

Reply:
This should definitely be `*const Self` (mirrors `&self` instead of `&mut self` in `get`'s signature)